cocli/cmd/corimSign.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2021 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/veraison/corim/corim"
+	cose "github.com/veraison/go-cose"
+)
+
+var (
+	corimSignCorimFile  *string
+	corimSignKeyFile    *string
+	corimSignOutputFile *string
+	corimSignMetaFile   *string
+)
+
+var corimSignCmd = NewCorimSignCmd()
+
+func NewCorimSignCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "sign",
+		Short: "create a signed CoRIM from an unsigned, CBOR-encoded CoRIM using the supplied key",
+		Long: `create a signed CoRIM from an unsigned, CBOR-encoded CoRIM using the supplied key
+
+	Sign the unsigned CoRIM unsigned-corim.cbor using the key in JWK format from
+	file key.jwk and save the resulting COSE Sign1 to signed-corim.cbor.  Read
+	the relevant CorimMeta information from file meta.json.
+	
+	  cli corim sign --file=unsigned-corim.cbor \
+	             --key=key.jwk \
+				 --meta=meta.json \
+				 --output=signed-corim.cbor
+	`,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := checkCorimSignArgs(); err != nil {
+				return err
+			}
+
+			// checkCorimSignArgs makes sure corimSignCorimFile is not nil
+			coseFile, err := sign(*corimSignCorimFile, *corimSignKeyFile,
+				*corimSignMetaFile, corimSignOutputFile)
+			if err != nil {
+				return err
+			}
+			fmt.Printf(">> %q signed and saved to %q\n", *corimSignCorimFile, coseFile)
+
+			return nil
+		},
+	}
+
+	corimSignCorimFile = cmd.Flags().StringP("file", "f", "", "an unsigned CoRIM file (in CBOR format)")
+	corimSignMetaFile = cmd.Flags().StringP("meta", "m", "", "CoRIM Meta file (in JSON format)")
+	corimSignKeyFile = cmd.Flags().StringP("key", "k", "", "signing key in JWK format")
+	corimSignOutputFile = cmd.Flags().StringP("output", "o", "", "name of the generated COSE Sign1 file")
+
+	return cmd
+}
+
+func checkCorimSignArgs() error {
+	if corimSignCorimFile == nil || *corimSignCorimFile == "" {
+		return errors.New("no CoRIM supplied")
+	}
+
+	if corimSignKeyFile == nil || *corimSignKeyFile == "" {
```

Comment:
Is there a mechanism to check whether a supplied key adheres to a JWK file format or not?